internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -919,3 +952,367 @@ func TestMachineSetReconciler_updateStatusResizedCondition(t *testing.T) {
+
+		return ns, cluster
+	}
+
+	teardown := func(t *testing.T, g *WithT, ns *corev1.Namespace, cluster *clusterv1.Cluster) {
+		t.Helper()
+
+		t.Log("Deleting the Cluster")
+		g.Expect(env.Delete(ctx, cluster)).To(Succeed())
+		t.Log("Deleting the namespace")
+		g.Expect(env.Delete(ctx, ns)).To(Succeed())
+	}
+
+	t.Run("Should sync machines with the in-place mutable changes", func(t *testing.T) {
+		g := NewWithT(t)
+		namespace, testCluster := setup(t, g)
+		defer teardown(t, g, namespace, testCluster)
+
+		replicas := int32(2)
+		version := "v1.25.3"
+		duration10s := &metav1.Duration{Duration: 10 * time.Second}
+		ms := &clusterv1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       "abc-123-ms-uid",
+				Name:      "ms-1",
+				Namespace: namespace.Name,
+				Labels: map[string]string{
+					"label-1": "true",
+				},
+			},
+			Spec: clusterv1.MachineSetSpec{
+				ClusterName: testCluster.Name,
+				Replicas:    &replicas,
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"machine-set-matching-label": "true",
+					},
+				},
+				Template: clusterv1.MachineTemplateSpec{
+					ObjectMeta: clusterv1.ObjectMeta{
+						Labels: map[string]string{
+							"machine-set-matching-label": "true",
+						},
+						Annotations: map[string]string{
+							"annotation-1": "true",
+							"precedence":   "MachineSet",
+						},
+					},
+					Spec: clusterv1.MachineSpec{
+						ClusterName: testCluster.Name,
+						Version:     &version,
+						Bootstrap: clusterv1.Bootstrap{
+							ConfigRef: &corev1.ObjectReference{
+								APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
+								Kind:       "GenericBootstrapConfigTemplate",
+								Name:       "ms-template",
+							},
+						},
+						InfrastructureRef: corev1.ObjectReference{
+							APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+							Kind:       "GenericInfrastructureMachineTemplate",
+							Name:       "ms-template",
+						},
+						NodeDrainTimeout:        duration10s,
+						NodeDeletionTimeout:     duration10s,
+						NodeVolumeDetachTimeout: duration10s,
+					},
+				},
+			},
+		}
+
+		fieldV1Map := map[string]interface{}{
+			"f:metadata": map[string]interface{}{
+				"f:name": map[string]interface{}{},
+			},
+		}
+		fieldV1, err := json.Marshal(fieldV1Map)
+		g.Expect(err).To(BeNil())
+		inPlaceMutatingMachine := &clusterv1.Machine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       "Machine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				UID:         "abc-123-uid",
+				Name:        "in-place-mutating-machine",
+				Namespace:   namespace.Name,
+				Labels:      map[string]string{},
+				Annotations: map[string]string{},
+				ManagedFields: []metav1.ManagedFieldsEntry{
+					{
+						Manager:    "manager",
+						Operation:  metav1.ManagedFieldsOperationUpdate,
+						FieldsType: "FieldsV1",
+						FieldsV1:   &metav1.FieldsV1{Raw: fieldV1},
+					},
+				},
```

Comment:
```suggestion
				Annotations: map[string]string{
					// This will be overwritten with the annotation from the MachineSet
					"precedence": "Machine",
				},
```
Let's drop managed fields here. Also adding the predence annotation, otherwise we can't really verify that it's overwritten with the one from MachineSet